Xamarin.Essentials/Screenshot/MediaFile.shared.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Xamarin.Essentials
+{
+    public class ScreenshotFile : ReadOnlyFile, IDisposable
+    {
+        public bool Disposed { get; private set; }
+
+        internal ScreenshotFile(string filePath)
+            : base(filePath, "image/png")
+        {
+        }
+
+#if !NETSTANDARD1_0
+        public Stream AsStream() => System.IO.File.Open(FullPath, FileMode.Open, FileAccess.Read);
+#endif
+
+        ~ScreenshotFile()
+        {
+            Dispose(true);
+        }
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+            Dispose(true);
+        }
+
+        protected virtual void Dispose(bool isDisposing)
+        {
+            if (Disposed)
+                return;
+#if !NETSTANDARD1_0
+            if (isDisposing)
+            {
+                try
+                {
+                    System.IO.File.Delete(FullPath);
```

Comment:
I am not sure how I feel about this just yet... I like the idea, and it is something that I didn't think of when I was working on the file picker. We do want to clean up after ourselves. But... it might be safer for the user to do it as they may be storing the value somewhere to read later. Thoughts?

Reply:
I really like this pattern, I usually use it to force ownership semantics on the consumer. The file might be deleted eventually (I have no idea how finalizers work on ios/android/tizen, but I assume they have the same "guarantees" for running as normal finalizers, which are basically none.) As such the user is forced to do something with the file and can not just leave an artifact lying aroung, even if he does not dispose of it. I would just annotate this behaviour in the comments.

Reply:
Good enough.